IPython/core/ultratb.py

```diff
@@ -101,10 +101,7 @@
 import tokenize
 import traceback
 
-try:  # Python 2
-    generate_tokens = tokenize.generate_tokens
-except AttributeError:  # Python 3
-    generate_tokens = tokenize.tokenize
+generate_tokens = tokenize.tokenize
```

Comment:
The comments appear to be wrong, `generate_tokens = tokenize.generate_tokens` work on Python 3; and this is likely the cause for the  test failures.